frontend/src/scenes/insights/insightLogic.tsx

```diff
@@ -306,8 +306,6 @@ export const insightLogic = kea<insightLogicType>({
                 setFilters: (state, { filters }) => cleanFilters(filters, state),
                 setInsight: (state, { insight: { filters }, options: { overrideFilter } }) =>
                     overrideFilter ? cleanFilters(filters || {}) : state,
-                loadInsightSuccess: (state, { insight }) =>
```

Comment:
FYI I'm reverting this as this seems to be causing unintended side effects (see failing tests).

Reply:
:+1: I also have some fixes on top of this, I'll push that too

Reply:
I wanted to add Jest tests for `reportInsightViewed` and the right actions WERE fired according to the `At timeout had received these actions:` list, but for whatever reason the Kea testing scaffolding was not matching the action in the list to the action in `toDispatchActions`... Anyway, refactored things a bit, should be good now